### ISSUE TITLE
When the window opens with maximization,the rounded does not take effect.

### ISSUE THREAD

**Issue by Young-exec** [flexagoon/rounded-window-corners#54]
https://github.com/user-attachments/assets/6ed27469-acde-4152-b9b5-aad3d2d20eda

OS:fedora41
Gnome version:46.5
extension version:7

When the window opens with maximization and you try to unmaximize it, the rounded corner effect does not work.Version3 doesn't have this problem.

**Comment by wxmup** [comment]
兄弟,你用的是mutter-performance吗?

**Comment by flexagoon** [comment]
I can't reproduce this. Does this happen even with all other extensions disabled?

**Comment by one1001** [comment]
Hello, happens to me as well. While on Fedora 40 and even after switching to Fedora 41. 

I've tested it by turning on off all other extensions in both 40 and 41 and the issue persists. 

It occurs to me on Firefox, I also use a lot of firefox PWAs from the PWA extension, and it happens on all of those, so it is very noticeable for me.

I've noticed that turning both settings for keeping rounded corners on when maximized and fullscreen solves the issue (but it leaves me with the ugly seethrough gaps on the corners when I maximize or go fullscreen, which I find ugly hahaha). 

If I turn the extension off and back on, the rounded corners come back. If I close the app and restart it, the rounded corners also come back.

I think it must have something to do with maximing/unmaximizing.
